routing-daemon/lib/openshift/routing/models/f5-icontrol-rest.rb

```diff
@@ -235,11 +184,87 @@ def delete_pool_alias pool_name, alias_str
     end
 
     def add_ssl pool_name, alias_str, ssl_cert, private_key
-      # no-opp
-    end 
+      # The LTM must be configured with a default client SSL profile for
+      # fallback for custom alias SSL to work with SNI.
+      # https://support.f5.com/kb/en-us/solutions/public/13000/400/sol13452.html
+
+      begin
+        # write temp cert and key
+        certfname = Tempfile.new('bigip-ssl-cert')
+        keyfname = Tempfile.new('bigip-ssl-key')
+
+        certfname.write(ssl_cert)
+        certfname.close
+        keyfname.write(private_key)
+        keyfname.close
+
+        # scp cert and to F5 LTM (requires ssh key to be in authorized_keys on the F5 LTM
+        @logger.debug("Copying certificate for alias #{alias_str} for pool #{pool_name} to LTM host")
+        result = `scp -i #{@ssh_private_key} #{certfname.path} admin@#{@host}:/var/tmp/#{alias_str}.crt`
+        result = `scp -i #{@ssh_private_key} #{keyfname.path} admin@#{@host}:/var/tmp/#{alias_str}.key`
+
+        @logger.debug("LTM cert to be installed /var/tmp/#{alias_str}.crt")
+        post(url: "https://#{@host}/mgmt/tm/sys/crypto/cert",
+               payload: {
+                 "command" => "install",
+                 "name" => "#{alias_str}-https-cert",
+                 "from-local-file" => "/var/tmp/#{alias_str}.crt"
+               }.to_json)
+
+        @logger.debug("LTM cert to be installed /var/tmp/#{alias_str}.key")
+        post(url: "https://#{@host}/mgmt/tm/sys/crypto/key",
+               payload: {
+                 "command" => "install",
+                 "name" => "#{alias_str}-https-key",
+                 "from-local-file" => "/var/tmp/#{alias_str}.key"
+               }.to_json)
+
+        @logger.debug("LTM creating client-ssl profile for  #{URI.escape(alias_str)}")
+        post(url: "https://#{@host}/mgmt/tm/ltm/profile/client-ssl",
+                 payload: {
+                   "name" => "#{URI.escape(alias_str)}-ssl-profile",
+                   "cert" => "#{URI.escape(alias_str)}-https-cert.crt",
+                   "key" => "#{URI.escape(alias_str)}-https-key.key",
+                   "serverName" => "#{URI.escape(alias_str)}"
```

Comment:
I suspect we don't really want a `URI.escape` here in the payload; we only need it in the `delete` calls in `remove_ssl` below because there the alias actually is part of the URI.